ooni/reporter.py

```diff
@@ -298,8 +299,13 @@ def createReport(self):
             self.agent = SOCKS5Agent(reactor, proxyEndpoint=proxyEndpoint)
 
         elif self.collectorAddress.startswith('https://'):
-            # XXX add support for securely reporting to HTTPS collectors.
-            log.err("HTTPS based collectors are currently not supported.")
+            # not sure if there's something else it needs.  Seems to work.
+            # Very difficult to get it to work with self-signed certs.
+            self.agent = Agent(reactor)
```

Comment:
Because of the difficulties in getting twisted 15 to work with self-signed certificates, I ended up adding our made-up CA cert to the system's set of trusted certificate authorities.  This is probably a better option than making twisted blindly trust a self-signed certificate.